flytekit/core/workflow.py

```diff
@@ -250,28 +215,52 @@ def _local_execute(self, ctx: FlyteContext, **kwargs) -> Union[Tuple[Promise], P
                 t = self._native_interface.inputs[k]
                 kwargs[k] = Promise(var=k, val=TypeEngine.to_literal(ctx, v, t, self.interface.inputs[k].type))
 
+        # The output of this will always be a combination of Python native values and Promises containing Flyte
+        # Literals.
         function_outputs = self.execute(**kwargs)
+
+        # First handle the empty return case.
+        # A workflow function may return a task that doesn't return anything
+        #   def wf():
+        #       return t1()
+        # or it may not return at all
+        #   def wf():
+        #       t1()
+        # In the former case we get the task's VoidPromise, in the latter we get None
         if (
```

Comment:
so in this case we do not assert that output specification matches the actual outputs? I think we should right?

Reply:
sorry i don't understand... can you elaborate?